api/inc/halt_exports.h

```diff
@@ -53,4 +53,42 @@ typedef enum {
     __THALTERROR_MAX /* always keep as the last element of the enum */
 } THaltError;
 
+/* A pointer to this structure will be given to halt_error() handler
+ * of the debug box driver. */
+typedef struct {
+    /* A basic exception stack frame that is always present with a valid stack. */
+    struct {
+        uint32_t r0;
+        uint32_t r1;
+        uint32_t r2;
+        uint32_t r3;
+        uint32_t r12;
+        uint32_t lr;
+        uint32_t pc;
+        uint32_t xpsr;
+    } stack_frame;
+
+    /* A few registers that may be useful for debug. */
```

Comment:
Moving the printing code into a separate file (should be used by both core and lib) is relatively a big task. We agreed to file a separate ticket for it.